etc-lite-patch <- v1.12.2

meowsbits opened the pull request:
Merges upstream (ethereum/go-ethereum) thru latest v1.12.2 release into the `etc-lite-patch` branch.

alexmylonas committed:
internal/ethapi: make NewAccount return EIP-55 format (#26973)

This change implements returning the address as EIP-55 encoded when creating a new account.

stephenfire committed:
rpc: more accurate checking of handler method signatures (#27287)

This changes the RPC server to ignore methods using *context.Context as parameter
and *error as return value type. Methods with such types would crash the server when
called.

fjl committed:
go.mod: upgrade gencodec (#27288)

zsfelfoldi committed:
beacon/types: add beacon chain data types (#27292)

* beacon/types: add beacon chain data types

* beacon/merkle: added comments

* go.mod: cleanups

---------

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

islishude committed:
go.mod: use`gopkg.in/yaml.v3` instead of `github.com/go-yaml/yaml` (#27295)

karalabe committed:
all: tie timestamp based forks to the passage of London (#27279)

holiman committed:
ethdb/pebble: prevent shutdown-panic (#27238)

One difference between pebble and leveldb is that the latter returns error when performing Get on a closed database, the former does a panic. This may be triggered during shutdown (see #27237)

This PR changes the pebble driver so we check that the db is not closed already, for several operations. It also adds tests to the db test-suite, so the previously implicit assumption of "not panic:ing at ops on closed database" is covered by tests.

islishude committed:
cmd/utils: do not check free disk space in dev mode (#27281)

fjl committed:
beacon/types: auto-generate SyncCommittee marshaling methods (#27296)

jsvisa committed:
rpc, internal/cmdtest: increase timeout in tests (#27083)

This change gives the cmd-tests have a bit more time to finish before getting forcibly torn down.

s1na committed:
graphql: upgrade UI to v2 (#27294)

Upgrades  graphiql to v2.4.4. The interface has become much nicer, and there are extra features like tabs, history, dark mode etc.

This change also now uses golang embed to bundle the resources.

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

chawin-a committed:
accounts/abi: add ErrorById (#27277)

Adds `ErrorById` lookup

minh-bq committed:
eth/tracers: fix flatCallTracer crasher (#27304)

FlatCallTracer had a crasher when it was passed `onlyTopCall: true` as config.
This PR ignores config fields inherited from the normal call tracer.

pcw109550 committed:
core/state: do not ignore null addr while iterative dump (#27320)

fixes bug which caused the zero-address to be ignored during an iterative state-dump.

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

holiman committed:
cmd/evm: make batched state-test execution possible (#27318)

implements the ability to run several state-tests in one instance. By not providing a statetest path to the `evm statetest` command, the path(s) will instead be read from `stdin`.

holiman committed:
rpc: change BlockNumber constant values to match ethclient (#27219)

ethclient accepts certain negative block number values as specifiers for the "pending",
"safe" and "finalized" block. In case of "pending", the value accepted by ethclient (-1)
did not match rpc.PendingBlockNumber (-2).

This wasn't really a problem, but other values accepted by ethclient did match the
definitions in package rpc, and it's weird to have this one special case where they don't.

To fix it, we decided to change the values of the constants rather than changing ethclient.
The constant values are not otherwise significant. This is a breaking API change, but we
believe not a dangerous one.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

jsvisa committed:
cmd: use errrors.New instead of empty fmt.Errorf (#27329)

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
ethclient,event: replace noarg fmt.Errorf with errors.New (#27334)

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
crypto: replace noarg fmt.Errorf with errors.New (#27333)

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
les, signer, light: replace noarg fmt.Errorf with errors.New (#27336)


Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
internal,tests: replace noarg fmt.Errorf with errors.New (#27335)

* internal: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

* tests: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

* tests: go autoimport

Signed-off-by: jsvisa <delweng@gmail.com>

* tests: go autoimport

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
eth,consensus: replace noarg fmt.Errorf with errors.New (#27330)

* eth: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

* consensus: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

holiman committed:
ethdb/pebble: fix NewBatchWithSize to set db (#27350)

holiman committed:
params: go-ethereum v1.12.0 stable

holiman committed:
params: begin v1.12.1 release cycle

jsvisa committed:
core,console: replace noarg fmt.Errorf with errors.New (#27332)

* core: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

* console: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

* core: go autoimport

Signed-off-by: jsvisa <delweng@gmail.com>

* core: dry

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
accounts: replace noarg fmt.Errorf with errors.New (#27331)

* accounts: replace noarg fmt.Errorf with errors.New

Signed-off-by: jsvisa <delweng@gmail.com>

* accounts: go autoimport

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
eth/filters: retrieve logs in async (#27135)

This change implements async log retrievals via feeding logs in channels, instead of returning slices. This is a first step to implement #15063.  

---------

Signed-off-by: jsvisa <delweng@gmail.com>
Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Sina Mahmoodi <1591639+s1na@users.noreply.github.com>

jsvisa committed:
internal/ethapi: don't return header size from rpc (#27347)

RPC methods `eth_getHeaderBy*` returned a size value which was meant for internal
processes. Please instead use `size` field returned by `eth_getBlockBy*` if you're interested
in the RLP encoded storage size of the block.

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
eth: split api.go into namespace based files (#27263)

This change splits up the multiple API functions / namespaces currently defined in the eth package into different per-namespace files.

jsvisa committed:
internal/ethapi: add more testcases for block/header rpc (#27325)

Signed-off-by: jsvisa <delweng@gmail.com>
Co-authored-by: Sina Mahmoodi <1591639+s1na@users.noreply.github.com>

gballet committed:
core/types: remove superfluous todo-comment (#27383)

jwasinger committed:
eth: make debug_StorageRangeAt take a block hash or number (#27328)

eth: make StorageRangeAt take a block hash or number

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

ucwong committed:
les, eth: fix typo in comment (#27369)

dbadoy committed:
event: move type fixation logic into Feed.init (#27249)

This is a minor optimization/refactoring of Feed.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

prestwich committed:
internal/ethapi: prevent unnecessary resource usage in eth_getProof implementation (#27310)

Deserialize hex keys early to shortcut on invalid input, and re-use the account storageTrie for each proof for each proof in the account, preventing repeated deep-copying of the trie.

Closes #27308

 --------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>

rjl493456442 committed:
miner: suspend miner if node is syncing (#27218)

Drop the notions of uncles, and disables activities while syncing

-  Disable activities (e.g. generate pending state) while node is syncing,
-  Disable empty block submission (but empty block is still kept for payload building),
-  Drop uncle notion since (ethash is already deprecated)

ucwong committed:
go.mod: golang.org/x upgrade (#27299)

go.mod:golang upgrade

karalabe committed:
consensus, core, eth/downloader, params: 4844 chain validation (#27382)

karalabe committed:
core, eth/downloader, params: validate blob tx bodies (#27392)

karalabe committed:
core, eth/downloader: validate blobtx.To at serialization time (#27393)

fjl committed:
p2p/discover: add Table configuration and Nodes method (#27387)

* p2p/discover: remove ReadRandomNodes

Even though it's public, this method is not callable by code outside of
package p2p/discover because one can't get a valid instance of Table.

* p2p/discover: add Table.Nodes

* p2p/discover: make Table settings configurable

In unit tests and externally developed cmd/devp2p test runs, it can be
useful to tune the timer intervals used by Table.

gballet committed:
core/state: move slot RLP encoding into the MPT implementation (#27000)

Continuing with a series of PRs to make the Trie interface more generic, this PR moves
the RLP encoding of storage slots inside the StateTrie and light.Trie implementations,
as other types of tries don't use RLP.

lochjin committed:
internal/web3ext: fix parameter count of miner_start (#27400)

rjl493456442 committed:
core/state: mark account as dirty when resetObject occurs (#27339)

This changes the journal logic to mark the state object dirty immediately when it
is reset. 

We're mostly adding this change to appease the fuzzer. Marking it dirty immediately
makes no difference in practice because accounts will always be modified by EVM
right after creation.

karalabe committed:
all: remove the Rinkeby testnet (#27406)

karalabe committed:
eth/downloader: drop beacon head updates if the syncer is restarting (#27397)

* eth/downloader: drop beacon head updates if the syncer is restarting

* eth/donwloader: v2 of the goroutine spike preventer

joohhnnn committed:
graphql: simplify tx resolve (#27285)

rjl493456442 committed:
core/state: clear out cached state data when reset occurs (#27376)

* core/state: remove cached snap data if reset occurs

* core/state: address comment from peter

* core/state: skip revert in case data is nil

MariusVanDerWijden committed:
core: 4844 opcode and precompile (#27356)

* core: crypto: implement BLOBHASH and pointEval precompile

* core: crypto: fixed nitpicks, moved precompile return value

* core/vm: fix review comments

rjl493456442 committed:
core/state/snapshot, core/types, eth: move account definition to type (#27323)

* core/state/snapshot, core/types, eth: move account definition to type

* core, eth: revert snapshot Account API change

holiman committed:
cmd/evm: make evm blocktest output logs if so instructed (#27396)

* cmd/evm: make evm blocktest output logs if so instructed

* Apply suggestions from code review

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>

---------

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>

dkatzan committed:
core/txpool: fix minor flaw in isGapped check (#27404)

This fix a minor implementation issue with the newly introduced isGapped function and it is described in ticket: https://github.com/ethereum/go-ethereum/issues/27401

palango committed:
les: remove unused var `AdvertiseProtocolVersions` (#27405)

karalabe committed:
core/txpool: make transaction validation reusable across packages (pools) (#27429)

* core/txpool: abstraction prep work for secondary pools (blob pool)

* core/txpool: leave subpool concepts to a followup pr

* les: fix tests using hard coded errors

* core/txpool: use bitmaps instead of maps for tx type filtering

jsvisa committed:
eth,core: add api debug_getTrieFlushInterval (#27303)

* core,eth: add api debug_getTrieFlushInterval

Signed-off-by: jsvisa <delweng@gmail.com>

* eth/api_debug: comment of SetTrieFlushInterval

Signed-off-by: jsvisa <delweng@gmail.com>

* Apply suggestions from code review

---------

Signed-off-by: jsvisa <delweng@gmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

s1na committed:
graphql: implement withdrawals (EIP-4895) (#27072)

implements withdrawals in graphql as per https://github.com/ethereum/execution-apis/pull/400

stephenfire committed:
rlp: use identical receiver names for encBuffer methods (#27430)

ThreeAndTwo committed:
accounts/keystore: handle error for invalid key in DecryptKey (#27432)


Co-authored-by: KAI <kaili@coinsummer.io>
Co-authored-by: Felix Lange <fjl@twurst.com>

jsvisa committed:
eth/filters: reuse handleLogs method for removed logs (#27438)

jwasinger committed:
miner: log number of withdrawals in block upon payload update (#27457)

hero5512 committed:
internal/ethapi: remove error return on RPCMarshalBlock (#27449)

rm error when marshal block to rpc type allen

mmsqe committed:
rpc: add limit for batch request items and response size (#26681)

This PR adds server-side limits for JSON-RPC batch requests. Before this change, batches
were limited only by processing time. The server would pick calls from the batch and
answer them until the response timeout occurred, then stop processing the remaining batch
items.

Here, we are adding two additional limits which can be configured:

- the 'item limit': batches can have at most N items
- the 'response size limit': batches can contain at most X response bytes

These limits are optional in package rpc. In Geth, we set a default limit of 1000 items
and 25MB response size.

When a batch goes over the limit, an error response is returned to the client. However,
doing this correctly isn't always possible. In JSON-RPC, only method calls with a valid
`id` can be responded to. Since batches may also contain non-call messages or
notifications, the best effort thing we can do to report an error with the batch itself is
reporting the limit violation as an error for the first method call in the batch. If a batch is
too large, but contains only notifications and responses, the error will be reported with
a null `id`.

The RPC client was also changed so it can deal with errors resulting from too large
batches. An older client connected to the server code in this PR could get stuck
until the request timeout occurred when the batch is too large. **Upgrading to a version
of the RPC client containing this change is strongly recommended to avoid timeout issues.**

For some weird reason, when writing the original client implementation, @fjl worked off of
the assumption that responses could be distributed across batches arbitrarily. So for a
batch request containing requests `[A B C]`, the server could respond with `[A B C]` but
also with `[A B] [C]` or even `[A] [B] [C]` and it wouldn't make a difference to the
client.

So in the implementation of BatchCallContext, the client waited for all requests in the
batch individually. If the server didn't respond to some of the requests in the batch, the
client would eventually just time out (if a context was used).

With the addition of batch limits into the server, we anticipate that people will hit this
kind of error way more often. To handle this properly, the client now waits for a single
response batch and expects it to contain all responses to the requests.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

freeman-jiang committed:
core/types: add "chainID" field to legacy tx JSON encoding (#27452)


Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

zhiqiangxu committed:
rpc: add method to test for subscription support (#25942)

This adds two ways to check for subscription support. First, one can now check
whether the transport method (HTTP/WS/etc.) is capable of subscriptions using
the new Client.SupportsSubscriptions method.

Second, the error returned by Subscribe can now reliably be tested using this
pattern:
    
    sub, err := client.Subscribe(...)
    if errors.Is(err, rpc.ErrNotificationsUnsupported) {
        // no subscription support
    }

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

pengin7384 committed:
node: remove unused error return from Attach (#27450)

node: Delete the unused error from return parameters of Node.Attach() func

s1na committed:
eth/tracers: refactor exporting js buffer (#27472)

We had to do this workaround because it wasn't possible to export typed arrays from
JS to []byte. This was added in dop251/goja@2352993, so we can use the better way now.

aaronbuchwald committed:
core: remove unnecessary log copy (#27475)

The logs in this function are pulled straight from disk in rawdb.ReadRawReceipts and 
also modified in receipts.DeriveFields, so removing the copy should be fine.

holiman committed:
core/types: add support for BlobTxType receipts (#27470)


Co-authored-by: Felix Lange <fjl@twurst.com>

cuishuang committed:
cmd/evm: fix typos in docs (#27478)

fix some typos

Signed-off-by: cui fliter <imcusg@gmail.com>

joohhnnn committed:
rlp: fix typos (#27484)

fix typos

Co-authored-by: john <yejiarui@123.com>

karalabe committed:
all: move main transaction pool into a subpool (#27463)

* all: move main transaction pool into a subpool

* go.mod: remove superfluous updates

* core/txpool: review fixes, handle txs rejected by all subpools

* core/txpool: typos

Commit:
build: use slices package for sorting (#27486)

lochjin committed:
core/txpool/legacypool: reheap the tx list if london fork not enabled (#27481)

This change ensures Reheap will be called even before the London fork activates.
Since Reheap would otherwise only be called through `SetBaseFee` after London,
the list would just keep growing if the fork was not enabled or not reached yet.

Commit:
p2p: use slices package for sorting (#27494)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
ethdb/dbtest: use slices package for sorting (#27491)

joohhnnn committed:
internal/jsre/deps: fix typos in web3.js (#27500)

hero5512 committed:
core/txpool/legacypool: remove redundant check for floatingRatio == 0 (#27477)

floatingRatio is a constant and always non-zero. So there is no need to
check for == 0.

holiman committed:
core/txpool/legacypool: handle missing head in reset (#27479)

Fixes #27301, a crash that could occur during txpool reorg handling.

zchee committed:
rpc: avoid use of cgo by hard-coding maxPathSize (#27447)

Package rpc uses cgo to find the maximum UNIX domain socket path 
length. If exceeded, a warning is printed. This is the only use of cgo in this
package. It seems excessive to depend on cgo just for this warning, so
we now hard-code the usual limit for Linux instead.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
accounts/keystore: use slices package for sorting (#27485)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
cmd/devp2p: use slices package for sorting (#27487)

joohhnnn committed:
cmd/evm: fix typos in documentation (#27501)

pengin7384 committed:
consensus/clique: fix typo in comment (#27503)

Commit:
metrics: use slices package for sorting (#27493)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
consensus/clique: use slices package for sorting (#27488)

Also adds Address.Less for sorting use in other packages.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
internal/ethapi, les: use slices package for sorting (#27492)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
tests, trie: use slices package for sorting (#27496)


Co-authored-by: Felix Lange <fjl@twurst.com>

MariusVanDerWijden committed:
go.mod: update kzg libraries to use big-endian (#27510)

* go.mod: update kzg libraries to use big-endian

* go.sum: ran go mod tidy

* core/testdata/precompiles: fix blob verification test

* core/testdata/precompiles: fix blob verification test

rjl493456442 committed:
trie: remove parameter 'fromLevel' in Prove (#27512)

This removes the feature where top nodes of the proof can be elided.
It was intended to be used by the LES server, to save bandwidth 
when the client had already fetched parts of the state and only needed
some extra nodes to complete the proof. Alas, it never got implemented
in the client.

hero5512 committed:
core/txpool/legacypool: narrow down the scope of the variable (#27471)

Variables discarded, included can be defined closer to their usage.

Co-authored-by: Martin Holst Swende <martin@swende.se>

Commit:
eth: use slices package for sorting (#27490)

Also adds Hash.Less method for sorting purposes.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

jsvisa committed:
internal/build: ignore some files in FindMainPackages (#27521)

Commit:
core: use slices package for sorting (#27489)


Co-authored-by: Felix Lange <fjl@twurst.com>

fjl committed:
core/txpool: remove use of errors.Join function (#27523)

his function was added in Go 1.20, but our compatibility target
is Go 1.19.

jwasinger committed:
internal/ethapi: use same state for each invocation within EstimateGas (#27505)

EstimateGas repeatedly executes a transaction, performing a binary search with multiple gas prices to determine proper pricing. Each call retrieves a new copy of the state (https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1017) . Because the pending/latest state can change during the execution of EstimateGas, this can potentially cause strange behavior (as noted here: https://github.com/ethereum/go-ethereum/pull/27502#issue-1761957009).

This PR modifies EstimateGas to retrieve the state once and use a copy of it for every call invocation it does.

rjl493456442 committed:
cmd, core/state, eth, tests, trie: improve state reader (#27428)

The state availability is checked during the creation of a state reader.

-    In hash-based database, if the specified root node does not exist on disk disk, then
    the state reader won't be created and an error will be returned.

-    In path-based database, if the specified state layer is not available, then the
    state reader won't be created and an error will be returned.

This change also contains a stricter semantics regarding the `Commit` operation: once it has been performed, the trie is no longer usable, and certain operations will return an error.

holiman committed:
ethdb/pebble: fsync for batch writes (#27522)

This is likely the culprit behind several data corruption issues, e.g. where data has been
written to the freezer, but the deletion from pebble does not go through due to process
crash.

prestwich committed:
internal/ethapi: quantity-encode storage keys in eth_getProof response (#27309)

This changes the eth_getProof method implementation to re-encode the requested
storage keys, canonicalizing them in the response. For backwards-compatibility reasons,
go-ethereum accepts non-canonical hex keys. Accepting them is fine, but we should
not mirror invalid inputs into the output.

Closes #27306

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Felix Lange <fjl@twurst.com>

Exca-DK committed:
p2p/discover: swap verification order in discv4 ping handler (#27532)

In all other UDPv4 methods, the deadline is checked first. It seems weird to me that ping is an exception. Deadline comparison is also less resource intensive.

Co-authored-by: Exca-DK <Exca-DK@users.noreply.github.com>

gballet committed:
core/state, light, trie: add UpdateContractCode to the Trie interface (#27476)

Verkle trees store the code inside the trie. This PR changes the interface to pass the code, as well as the dirty flag to tell the trie package if the code is dirty and needs to be updated. This is a no-op for the MPT and the odr trie.

fbac committed:
crypto/secp256k1: define NDEBUG only if not defined (#27550)

pengin7384 committed:
common/math: fix typo in comment (#27561)

jsvisa committed:
eth/protocols: put protocols in order (#27543)

Signed-off-by: jsvisa <delweng@gmail.com>

puhtaytow committed:
cmd/geth: error message should not be capitalised (#27549)

error message should not be capitalized / consistency

rjl493456442 committed:
ethdb/pebble: use sync mode for pebble writes (#27615)

MariusVanDerWijden committed:
core/vm: move TSTORE,TLOAD to correct opcode nums (#27613)

* core/vm: move TSTORE,TLOAD to correct opcode nums

* core/vm: cleanup

pengin7384 committed:
common/lru: add test case for BasicLRU.Peek (#27559)

aaronbuchwald committed:
core: log message if DeriveFields returns error (#27530)

This error will never happen, but if it ever does happen, we will at least see it.

gballet committed:
trie, core/state: revert error removal in (*state.Trie).Commit (#27544)

* trie, core/state: revert error removal in (*state.Trie).Commit

* Gary's nitpick :)

Co-Authored-By:  rjl493456442 <garyrong0905@gmail.com>

---------

Co-authored-by: rjl493456442 <garyrong0905@gmail.com>

hero5512 committed:
core/txpool/legacypool: fix typo (#27620)

MariusVanDerWijden committed:
eth: send big transactions by announce/retrieve only (#27618)

* eth: send big transactions by announce/retrieve only

* Update eth/handler.go

Co-authored-by: Martin Holst Swende <martin@swende.se>

* eth: remove superfluous bracket

* eth: add whitespace

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

gballet committed:
params, trie: add verkle fork management + upgrade go-verkle (#27464)

* params, trie: add verkle fork management + upgrade go-verkle

* remove the two verkle files

* core, eth, params: add missing function

* Gary's feedback

* remove trie/utils/verkle.go

* add verkle block override

---------

Co-authored-by: Gary Rong <garyrong0905@gmail.com>

gballet committed:
core/state, light, les: make signature of ContractCode hash-independent (#27209)

* core/state, light, les: make signature of ContractCode hash-independent

* push current state for feedback

* les: fix unit test

* core, les, light: fix les unittests

* core/state, trie, les, light: fix state iterator

* core, les: address comments

* les: fix lint

---------

Co-authored-by: Gary Rong <garyrong0905@gmail.com>

oiooj committed:
node: fix listening on IPv6 address (#27628) (#27635)

rjl493456442 committed:
cmd, core, eth, graphql, trie: no persisted clean trie cache file (#27525)

The clean trie cache is persisted periodically, therefore Geth can
quickly warmup the cache in next restart.

However it will reduce the robustness of system. The assumption is
held in Geth that if the parent trie node is present, then the entire
sub-trie associated with the parent are all prensent.

Imagine the scenario that Geth rewinds itself to a past block and
restart, but Geth finds the root node of "future state" in clean
cache then regard this state is present in disk, while is not in fact.

Another example is offline pruning tool. Whenever an offline pruning
is performed, the clean cache file has to be removed to aviod hitting
the root node of "deleted states" in clean cache.

All in all, compare with the minor performance gain, system robustness
is something we care more.

Exca-DK committed:
ethdb/leveldb: update leveldb metrics collection (#27643)

This removes text parsing in leveldb metrics collection code. All metrics
can now be accessed through the stats API provided by leveldb.

We also add new gauge-typed metrics that count the number of tables at each level.

---------

Co-authored-by: Exca-DK <Exca-DK@users.noreply.github.com>
Co-authored-by: Gary Rong <garyrong0905@gmail.com>
Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
core: stop chain when tests are finished (#27660)

core (blockchain_test): add chain.Stop() to tests

joohhnnn committed:
signer/core: show helptext when clef rejects signing (#27538)



Co-authored-by: Martin Holst Swende <martin@swende.se>

jwasinger committed:
cmd/geth: implement dev mode for post-merge (#27327)

This change adds back the 'geth --dev' mode of operation, using a cl-mocker. 

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: rjl493456442 <garyrong0905@gmail.com>
Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

lightclient committed:
p2p, p2p/discover: add dial metrics (#27621)

This PR adds metrics for p2p dialing, which gives us visibility into the quality of the dial 
candidates  returned by our discovery methods.

holiman committed:
core: remove outdated tests (#27662)

Back before #27178 , we spun up a number of ethash verifiers to verify headers. So we also had tests to ensure that we were indeed able to abort verification even if we had multiple workers running.

With PR #27178, we removed the parallelism in verification, and these tests are now failing, since we now just sequentially fire away the results as fast as possible on one routine.

This change removes the (sometimes failing) tests

jwasinger committed:
miner: remove unused commented code (#27664)

holiman committed:
core/vm: define cancun + enable 1153 (tstore/tload) in cancun (#27663)

charles-cooper committed:
core/vm: implement EIP-5656, mcopy instruction (#26181)

Implements [EIP 5656](https://eips.ethereum.org/EIPS/eip-5656), MCOPY instruction, and enables it for Cancun. 

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

lightclient committed:
eth: better active protocol handler tracking (#27665)

Fixes an issue where waitgroups were used erroneously, which could lead to waitgroup being added to while wait was already invoked.

jsvisa committed:
internal/blocktest: add package for shared test code (#27270)

rjl493456442 committed:
core/state: value diff tracking in StateDB (#27349)

This change makes the StateDB track the state key value diff of a block transition.
We already tracked current account and storage values for the purpose of updating
the state snapshot. With this PR, we now also track the original (pre-transition) values
of accounts and storage slots.

lightclient committed:
cmd/utils, p2p: clean up discovery setup (#27518)

This simplifies the code that initializes the discovery a bit, and
adds new flags for enabling/disabling discv4 and discv5 separately.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

Mskxn committed:
console: add cleanup to avoid leaks in newTester (#27695)

fjl committed:
go.mod: upgrade github.com/karalabe/usb to fix build warning (#27698)

rjl493456442 committed:
core/rawdb: support freezer batch read with no size limit (#27687)

This change adds the ability to perform reads from freezer without size limitation. This can be useful in cases where callers are certain that out-of-memory will not happen (e.g. reading only a few elements). 

The previous API was designed to behave both optimally and secure while servicing a request from a peer, whereas this change should _not_ be used when an untrusted peer can influence the query size.

fjl committed:
p2p/discover: filter bootnodes by netrestrict (#27701)

This prevents an issue where the node would attempt to contact the bootstrap
nodes even if they weren't contained in the netrestrict list.

cuishuang committed:
accounts, cmd/evm: fix docstrings (#27703)

fix function name in comment

Signed-off-by: cui fliter <imcusg@gmail.com>

sandakersmann committed:
build: upgrade -dlgo version to Go 1.20.6 (#27708)

jsvisa committed:
eth, miner: rm redundant type declare (#27713)

* eth: rm redundant type from array, slice, or map

* miner: rm redundant type from array, slice, or map

s1na committed:
graphql: fix race in withdrawals test (#27706)

graphql: fix race in test

jsvisa committed:
internal: early exit if tx for getTxReceipt not found (#27712)

internal/ethapi: fast exit if tx notfound

jsvisa committed:
internal/ethapi: add more testcase for GetTransactionReceipt (#27527)

* internal/ethapi: testBackend reuse the same db

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: implment GetTransaction

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: implement GetReceipts

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: insert receipts and setup txlookup

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add simple success tx receipt test

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add case create contract

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add contract call receipt

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add tx notfound

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add dynamic fee testcase

* internal/ethapi: add accessList receipt

* internal/ethapi: no need to insert receipt chain, no error

* internal/ethapi: use HeaderByHash instead

* internal/ethapi: add one more case

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
eth/tracers: add ReturnData in the tracer's response (#27704)

The struct logger (or opcode tracer) was missing the return data field even
if this was explicitly enabled by user via `"enableReturnData": true` in the config.
This PR fixes this issue.

Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

ryanschneider committed:
eth/tracers/native: panic on memory read in prestateTracer (#27691)


Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

ucwong committed:
go.sum: go mod tidy (#27717)

ucwong committed:
core/blockchain: fast to snap in comments (#27722)

dbadoy committed:
cmd/bootnode, p2p: support for alternate mapped ports (#26359)

This changes the port mapping procedure such that, when the requested port is unavailable
an alternative port suggested by the router is used instead.

We now also repeatedly request the external IP from the router in order to catch any IP changes.

Co-authored-by: Felix Lange <fjl@twurst.com>

omerfirmak committed:
metrics: NilTimer should still run the function to be timed (#27723)

jsvisa committed:
cmd/geth: use automaxprocs to apply cpu quota correctly (#27506)

It is usually best to set GOMAXPROCS to the number of available CPU cores. However, setting
it like that does not work well when the process is quota-limited to a certain number of CPUs.
The automaxprocs library configures GOMAXPROCS, taking such limits into account.

omerfirmak committed:
metrics: NilResettingTimer.Time should execute the timed function (#27724)

yuriechan committed:
accounts/abi/bind/backends: fix goroutine leak in unit test (#27705)

jwasinger committed:
core:  replace instances of 'suicide' with 'selfdestruct' to improve code consistency. (#27716)


---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

ucwong committed:
core/vm: use types.EmptyCodeHash (#27729)

holiman committed:
core, tests: EIP-4844 transaction processing logic (#27721)

This updates the reference tests to the latest version and also adds logic
to process EIP-4844 blob transactions into the state transition. We are now
passing most Cancun fork tests.

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>
Co-authored-by: Felix Lange <fjl@twurst.com>

jwasinger committed:
core/state, core/vm: implement EIP 6780 (#27189)

EIP-6780: SELFDESTRUCT only in same transaction

>     SELFDESTRUCT will recover all funds to the caller but not delete the account, except when called in the same transaction as creation

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

zsfelfoldi committed:
les: remove obsolete code related to PoW header syncing (#27737)

This change removes PoW header syncing related code from LES and also deletes 
duplicated packages les/catalyst, les/downloader and les/fetcher. These package copies
were created because people wanted to make changes in their eth/ counterparts, but weren't
able to adapt LES code to the API changes.

fjl committed:
beacon/engine, eth/catalyst: EIP-4844 updates for the engine API (#27736)

This is a spin-out from the EIP-4844 devnet branch, containing just the Engine API modifications
and nothing else. The newPayloadV3 endpoint won't really work in this version, but we need the
data structures for testing so I'd like to get this in early.

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>

pgrimaud committed:
cmd/evm/testdata: fix typos in docs (#27742)

MariusVanDerWijden committed:
ethclient/gethclient: gofmt -s (#27762)

rjl493456442 committed:
all: expose block number information to statedb (#27753)

* core/state: clean up

* all: add block number infomration to statedb

* core, trie: rename blockNumber to block

ucwong committed:
cmd/geth: atomic types used (#27756)

ucwong committed:
cmd/devp2p: atomic types used (#27755)

jtraglia committed:
crypto/kzg4844: do lazy init in all ckzg funcs (#27679)

* crypto/kzg4844: remove unnecessary init call & fix typo

* Fix kzg4844 tests/benchmarks

* Make init lazy & revert changes to tests

Commit:
core/types: deepcopy ExcessDataGas, DataGasUsed (#27767)

karalabe committed:
core/txpool/blobpool: 4844 blob transaction pool (#26940)

* core/blobpool: implement txpool for blob txs

* core/txpool: track address reservations to notice any weird bugs

* core/txpool/blobpool: add support for in-memory operation for tests

* core/txpool/blobpool: fix heap updating after SetGasTip if account is evicted

* core/txpool/blobpool: fix eviction order if cheap leading txs are included

* core/txpool/blobpool: add note as to why the eviction fields are not inited in reinject

* go.mod: pull in inmem billy form upstream

* core/txpool/blobpool: fix review commens

* core/txpool/blobpool: make heap and heap test deterministic

* core/txpool/blobpool: luv u linter

* core/txpool: limit blob transactions to 16 per account

* core/txpool/blobpool: fix rebase errors

* core/txpool/blobpool: luv you linter

* go.mod: revert some strange crypto package dep updates

fjl committed:
core/types: add 4844 data gas fields in Receipt (#27743)

* core/types: add data gas fields in Receipt

* core/types: use BlobGas method of tx

* core: fix test

* core/types: fix receipt tests, add data gas used field test

---------

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
core/txpool/blobpool: fix a merge conflict from a package rename (#27790)

karalabe committed:
all: rename dataGas to blobGas (#27789)

MariusVanDerWijden committed:
core/types: put header fields in correct order (#27791)

marioevz committed:
core/types: fix receipt blob fields marshaling (#27793)

islishude committed:
cmd/geth: import package catalyst once (#27803)

islishude committed:
eth: conver if-else-if chain to tagged switch (#27816)

jsvisa committed:
core: remove pointless check for excessBlobGas nilness (#27797)

* core: check excessBlobGas in front

Signed-off-by: jsvisa <delweng@gmail.com>

* core: no need to manual panic

Signed-off-by: jsvisa <delweng@gmail.com>

* core: no comment

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

rjl493456442 committed:
core, trie: track state change set with account address (#27815)

ZigBalthazar committed:
cmd/bootnode: defer udp connection close (#27787)

jsvisa committed:
all: replace data gas to blob gas in comments (#27825)

* eth: excessDataGas -> excessBlobGas

* consensus: data gas -> blob gas

* core: data gas -> blob gas

* params: data gas -> blob gas

holiman committed:
tests: make tests pass on windows/386 (#27821)

* tests: split up state test execution

* Revert "tests: split up state test execution"

This reverts commit 96017c248c85d24e93ad013a2bbe8b38c99327c0.

* build: bump test timeout to 20 minutes

rjl493456442 committed:
core/txpool/blobpool: remove error log when finalized block is nil (#27822)

* core/txpool/blobpool: remove error log when finalized block is nil

* core/txpool/blobpool: take peter's suggestion

jsvisa committed:
consensus/misc: move eip1559 into a package (#27828)

* consensus/misc: move eip1559 as a sub directory

Signed-off-by: jsvisa <delweng@gmail.com>

* consensus/misc: package name

Signed-off-by: jsvisa <delweng@gmail.com>

* all: eip1559

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
cmd/evm: set ExcessBlobGas from env (#27796)

Sets the `currentExcessBlobGas` from env, alternatively calculates it based on `parentExcessBlobGas` and `parentBlobGasUsed`. It then emits the `currentExcessBlobGas` and `currentBlobGasUsed` into the output, to be used as parent-values for a future iteration. 

Closes #27785
Closes #27783

---------

Signed-off-by: jsvisa <delweng@gmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

ucwong committed:
cmd/bootnode: fix timer leak (#27754)

rjl493456442 committed:
all: implement path-based state scheme (#25963)

* all: implement path-based state scheme

* all: edits from review

* core/rawdb, trie/triedb/pathdb: review changes

* core, light, trie, eth, tests: reimplement pbss history

* core, trie/triedb/pathdb: track block number in state history

* trie/triedb/pathdb: add history documentation

* core, trie/triedb/pathdb: address comments from Peter's review

Important changes to list:

- Cache trie nodes by path in clean cache
- Remove root->id mappings when history is truncated

* trie/triedb/pathdb: fallback to disk if unexpect node in clean cache

* core/rawdb: fix tests

* trie/triedb/pathdb: rename metrics, change clean cache key

* trie/triedb: manage the clean cache inside of disk layer

* trie/triedb/pathdb: move journal function

* trie/triedb/path: fix tests

* trie/triedb/pathdb: fix journal

* trie/triedb/pathdb: fix history

* trie/triedb/pathdb: try to fix tests on windows

* core, trie: address comments

* trie/triedb/pathdb: fix test issues

---------

Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

jsvisa committed:
cmd/geth: disable automaxprocs log (#27814)


Co-authored-by: Felix Lange <fjl@twurst.com>

ucwong committed:
p2p: use atomic types (#27764)


Co-authored-by: Felix Lange <fjl@twurst.com>

sandakersmann committed:
build: upgrade -dlgo version to Go 1.20.7 (#27835)

karalabe committed:
eth/protocols/snap: fix batch writer when resuming an aborted sync (#27842)

fjl committed:
core/types: fix immutability guarantees in Block (#27844)

This change rearranges the accessor methods in block.go and fixes some minor issues with
the copy-on-write logic of block data. Fixed issues:

- Block.WithWithdrawals did not create a shallow copy of the block.

- Block.WithBody copied the header unnecessarily, and did not preserve withdrawals.

However, the bugs did not affect any code in go-ethereum because blocks are *always*
created using NewBlockWithHeader().WithBody().WithWithdrawals()

Commit:
all: remove trailing whitespace (#27741)

fjl committed:
core/types: support yParity field in JSON transactions (#27744)

This adds support for the "yParity" field in transaction objects returned by RPC
APIs. We somehow forgot to add this field even though it has been in the spec for
a long time.

ucwong committed:
log: use atomic types (#27763)


Co-authored-by: Felix Lange <fjl@twurst.com>

chrischarlesharrison committed:
internal/ethapi, graphql: correct comments about gas price logic (#27752)

ucwong committed:
internal: use atomic type (#27858)

ucwong committed:
accounts: use atomic type (#27857)

amintalebi committed:
internal/ethapi: add state override to estimateGas (#27845)

ucwong committed:
p2p/simulations/examples: use atomic.Int64 (#27861)

holiman committed:
graphql: avoid greedy allocation (#27873)

Fixes a graphql-dos

---------

Co-authored-by: Sina Mahmoodi <1591639+s1na@users.noreply.github.com>
Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

holiman committed:
core/vm: update 4844 - point evaluation precompile address (#27874)

fjl committed:
p2p: move ping handling into pingLoop goroutine (#27887)

Moving the response sending there allows tracking all peer goroutines
in the peer WaitGroup.

jsvisa committed:
graphql: add yParity field for transactions (#27882)


Co-authored-by: Felix Lange <fjl@twurst.com>

gballet committed:
core/state: move UpdateContractCode before the trie hash is computed (#27853)

Context: The UpdateContractCode method was introduced for the state storage commitment
schemes that include the whole code for their commitment computation. It must therefore be called
before the root hash is computed at the end of IntermediateRoot.

This should have no impact on the MPT since, in this context, the method is a no-op.

fjl committed:
params: release go-ethereum v1.12.1

fjl committed:
params: begin v1.12.2 release cycle

dependabot[bot] committed:
deps: update supranational/blst to 0.3.11 (#27890)

build(deps): bump github.com/supranational/blst

Bumps [github.com/supranational/blst](https://github.com/supranational/blst) from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11.
- [Release notes](https://github.com/supranational/blst/releases)
- [Commits](https://github.com/supranational/blst/commits/v0.3.11)

---
updated-dependencies:
- dependency-name: github.com/supranational/blst
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

imulmat4 committed:
core/txpool/blobpool: fix metrics name for prometheus export (#27901)

jsvisa committed:
ethdb/leveldb: support more than 7 levels in metrics (#27904)

karalabe committed:
all: update golang/x/ext and fix slice sorting fallout (#27909)

The Go authors updated golang/x/ext to change the function signature of the slices sort method. 
It's an entire shitshow now because x/ext is not tagged, so everyone's codebase just 
picked a new version that some other dep depends on, causing our code to fail building.

This PR updates the dep on our code too and does all the refactorings to follow upstream...

fjl committed:
go.mod, build: upgrade c-kzg-4844 (#27907)

This upgrades to the latest release of ckzg, and also attempts to fix some blst-related
build errors that occur on launchpad.net.

fjl committed:
internal/build: apply -ubuntu to env (#27910)

fjl committed:
params: release go-ethereum v1.12.2

meowsbits committed:
Merge tag 'v1.12.2' into etc-lite-patch

Conflicts:
      cmd/geth/main.go
      cmd/utils/flags.go
      consensus/ethash/algorithm.go
      consensus/ethash/algorithm_test.go
      consensus/ethash/consensus.go
      consensus/ethash/ethash.go
      consensus/ethash/sealer.go
      consensus/ethash/sealer_test.go
      core/forkid/forkid.go
      eth/backend.go
      eth/catalyst/api_test.go
      eth/ethconfig/config.go
      go.sum
      internal/ethapi/api.go
      les/client.go
      miner/worker.go
      params/config.go

meowsbits committed:
consensus/misc: re-add file, etc likes verifying block hashes occassionally

Date: 2023-08-30 06:22:05-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2023-08-30 06:25:45-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils,eth/ethconfig: re-install Ethash config stuff

Date: 2023-08-30 06:26:28-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: whitespace

Date: 2023-08-30 06:28:10-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "miner: suspend miner if node is syncing (#27218)"

This reverts commit d4961881d7c92603f591f9cb8c705d00d8cbdfc0.

 Conflicts:
	miner/worker.go
	miner/worker_test.go

meowsbits committed:
Revert "les: remove obsolete code related to PoW header syncing (#27737)"

This reverts commit d4d88f9bce13ca9310bf28f5f26ea9f1915ba90d.

On branch etc-lite-patch-dev_with-reverts-from-chris
Changes to be committed:
      modified:   cmd/geth/main.go
      modified:   cmd/utils/flags.go
      modified:   eth/ethconfig/config.go
      modified:   eth/ethconfig/gen_config.go
      modified:   les/api_backend.go
      modified:   les/api_test.go
      new file:   les/catalyst/api.go
      new file:   les/catalyst/api_test.go
      modified:   les/client.go
      modified:   les/client_handler.go
      new file:   les/downloader/api.go
      new file:   les/downloader/downloader.go
      new file:   les/downloader/downloader_test.go
      new file:   les/downloader/events.go
      new file:   les/downloader/metrics.go
      new file:   les/downloader/modes.go
      new file:   les/downloader/peer.go
      new file:   les/downloader/queue.go
      new file:   les/downloader/queue_test.go
      new file:   les/downloader/resultstore.go
      new file:   les/downloader/statesync.go
      new file:   les/downloader/testchain_test.go
      new file:   les/downloader/types.go
      new file:   les/fetcher.go
      new file:   les/fetcher/block_fetcher.go
      new file:   les/fetcher/block_fetcher_test.go
      new file:   les/fetcher_test.go
      modified:   les/handler_test.go
      modified:   les/odr_test.go
      modified:   les/peer.go
      new file:   les/pruner.go
      new file:   les/pruner_test.go
      modified:   les/request_test.go
      modified:   les/retrieve.go
      new file:   les/sync.go
      new file:   les/sync_test.go
      modified:   les/test_helper.go
      new file:   les/ulc.go
      new file:   les/ulc_test.go

Changes not staged for commit:
      modified:   tests/testdata (new commits)

meowsbits committed:
consensus/beacon,consensus/clique,consensus,consensus/ethash,eth/ethconfig: reinstall ethash and related consensus engine params

Date: 2023-08-30 07:35:12-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/forkid: "sort" imported and not used) (typecheck)

Date: 2023-08-30 07:44:33-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/misc: undefined: misc.VerifyForkHashes

Date: 2023-08-30 07:49:16-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: not enough arguments in call to bc.engine.VerifyHeader

Date: 2023-08-30 07:52:54-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: not enough arguments in call to hc.engine.VerifyHeaders

Date: 2023-08-30 07:55:02-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: cannot use &txpool.LazyTransaction{…} (value of type *txpool.LazyTransaction) as *github.com/ethereum/go-ethereum/core/types.Transaction value in argument to append

Date: 2023-08-30 08:15:20-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: undefined: ethash

Date: 2023-08-30 08:16:28-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: re-install ethashConfig defaults

Date: 2023-08-30 08:22:26-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: remove checkpoint config; go-ethereum no longer has checkpoints

Date: 2023-08-30 08:22:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: not enough arguments in call to h.chain.Engine().VerifyHeader

Date: 2023-08-30 08:23:48-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: not enough arguments in call to ethconfig.CreateConsensusEngine

Date: 2023-08-30 08:25:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: config.Checkpoint undefined

Date: 2023-08-30 08:26:42-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: not enough arguments in call to engine.VerifyHeader

Date: 2023-08-30 08:27:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth: s/SetGasPrice/SetGasTip/

Date: 2023-08-30 08:32:23-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: threaded mining

Date: 2023-08-30 08:32:32-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: not enough arguments in call to api.e.StartMining

Date: 2023-08-30 08:33:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner/stress/clique: not enough arguments in call to node.StartMining

Date: 2023-08-30 08:35:47-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst: not enough arguments in call to engine.ExecutableDataToBlock

Date: 2023-08-30 08:36:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils,miner: cfg.Notify undefined

TODO I've reinstalled these fields in the
miner config, but don't see what they actually
do. Even looking at historical code its not
apparent. So this may be a broken config, but
we'll see.

Date: 2023-08-30 08:55:32-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: not enough arguments in call to ethconfig.CreateConsensusEngine

Date: 2023-08-30 08:58:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth: undefined: utils.MinerNoVerifyFlag

Date: 2023-08-30 13:27:51-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: not enough arguments in call to engine.VerifyHeaders

Date: 2023-08-30 13:30:55-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: not enough arguments in call to blockchain.engine.VerifyHeader

Date: 2023-08-30 13:33:32-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
light: not enough arguments in call to lightchain.engine.VerifyHeader

Date: 2023-08-30 13:34:55-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: File is not goimports-ed (goimports)

Date: 2023-08-30 13:36:30-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool,miner: b.txPool.AddLocals undefined (type *txpool.TxPool has no field or method AddLocals) (typecheck)

Date: 2023-08-30 13:52:25-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool,core/types: git checkout v1.12.2 -- core/txpool core/types

Date: 2023-08-31 06:04:39-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: cannot use pendingTxs (variable of type []*github.com/ethereum/go-ethereum/core/types.Transaction) as []*txpool.Transaction value in argument to backend.txPool.Add

Date: 2023-08-31 06:04:45-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: re-implement seal verification in verifyHeader

This fixes the failing tests at TestHeaderVerification

Date: 2023-08-31 06:45:22-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethclient/gethclient: re-install Ethash config = ModeFake for test

This fixes the failing TestGethClient which
was complaining about invalid mix digest.

Date: 2023-08-31 07:11:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethclient: re-install ethash config pow mode = Fake

Fixes failing TestEthClient

Date: 2023-08-31 07:20:00-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
graphql: re-install ethash pow mode = fake

Fixes failing TestGraphQLBlockSerialization

Date: 2023-08-31 07:25:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst: re-install ethash fake mode for tests

Date: 2023-08-31 07:31:59-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: re-add chain side event handling

This fixes TestStreamUncleBlock which a new test.
Still failing are two tests, both RegenerateMiningBlock...

Date: 2023-08-31 08:15:18-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: delete failing Regenerate tests

These tests (ethash and clique) were both
failing because the third and last event
were not received; apparently the txpool.Add
was supposed to trigger that at one point.
But anymore they don't trigger the event,
and I'm not even sure how they were to
have done so.
Upstream ethereum/go-ethereum at v1.12.2
DOES NOT have these tests (they were unique
to this branch), so I'm just going to remove them
since we're trying to stay close to upstream
anyways.

Date: 2023-08-31 09:11:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr commented on cmd/geth/main.go:
```diff
@@ -442,7 +442,8 @@ func startNode(ctx *cli.Context, stack *node.Node, backend ethapi.Backend, isCon
 		gasprice := flags.GlobalBig(ctx, utils.MinerGasPriceFlag.Name)
-		ethBackend.TxPool().SetGasPrice(gasprice)
+		// ethBackend.TxPool().SetGasPrice(gasprice)
+		ethBackend.TxPool().SetGasTip(gasprice)
```
Duplicate line, one has to be removed

ziogaschr commented on cmd/utils/flags.go:
```diff
@@ -568,11 +568,6 @@ var (
 		Value:    ethconfig.Defaults.Miner.Recommit,
 		Category: flags.MinerCategory,
 	}
-	MinerNoVerifyFlag = &cli.BoolFlag{
```
What if we keep this flag as well for ethash?

meowsbits replied:
SGTM, I removed it in a flurry.

meowsbits replied:
[2ddfa04](https://github.com/etclabscore/ethereum.go-ethereum/pull/18/commits/2ddfa042b2de815273061ba254390e4eda4029fe) re-installs it.

ziogaschr commented on core/txpool/subpool.go:
```diff
@@ -104,7 +104,7 @@ type SubPool interface {
 	// Add enqueues a batch of transactions into the pool if they are valid. Due
 	// to the large transaction churn, add may postpone fully integrating the tx
 	// to a later point to batch multiple ones together.
-	Add(txs []*Transaction, local bool, sync bool) []error
```
Want we break support of blobTXs (and bloppool) if we change the type from `txpool.Transaction` to `types.Transaction`?

Txpool at this point wants to just be a wrapper of its subpools, while `Locals` is a notion of the legacypool and pre-merge chains.

ziogaschr replied:
You figured it out in this commit https://github.com/etclabscore/ethereum.go-ethereum/pull/18/commits/3f44042b6ca4e418d5cdeb5e4b3200e9cd24127b already. :)

meowsbits committed:
cmd/geth: remove spurious duplicate lines

Date: 2023-09-01 07:00:50-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth,cmd/utils,miner: re-install miner flag configs

Install miner flags again -- noverify and notify
values are passed to the remote sealer.
